### PR TITLE
Switch to new search API

### PR DESCRIPTION
**Pull request by mdrohmann**
https://www.pivotaltracker.com/story/show/174961646

### CODE REVIEW

**Comment by mdrohmann** [comment]
I also a follow-up story to improve the UI/UX of `state packages search`: https://www.pivotaltracker.com/story/show/175423026

File: pkg/platform/model/inventory.go [hunk 1]
```diff
@@ -56,12 +55,15 @@ func IngredientByNameAndVersion(language, name, version string) (*IngredientAndV
...
-			return &IngredientAndVersion{
-				ingredient.V1IngredientAndVersion,
-				ingredient.Namespace,
-			}, nil
+
+		for _, ver := range ingredient.Versions {
+			v := ver.Version
+			if v == version {
+				return &IngredientAndVersion{
+					ingredient.V1SearchIngredientsResponseIngredientsItems, v,
```
Comment (MDrakos) [inline pkg/platform/model/inventory.go:63]: Nit: I was wondering where the version was until I saw it hiding on this line.

File: pkg/platform/model/inventory.go [hunk 2]
```diff
@@ -174,23 +152,25 @@ func searchIngredientsNamespace(limit int, namespace, language, name string) ([]
...
+	params.SetQ(name)
+	params.SetNamespaces(namespaceAndLanguage)
 	params.SetLimit(&lim)
 	params.SetHTTPClient(retryhttp.DefaultClient.StandardClient())
 
-	results, err := client.GetNamespaceIngredients(params, authentication.ClientAuth())
+	results, err := client.SearchIngredients(params, authentication.ClientAuth())
 	if err != nil {
-		if gniErr, ok := err.(*inventory_operations.GetNamespaceIngredientsDefault); ok {
+		if gniErr, ok := err.(*inventory_operations.SearchIngredientsDefault); ok {
```
Comment (MDrakos) [inline pkg/platform/model/inventory.go:163]: Nit: this is not a `gniErr` anymore

File: pkg/platform/model/inventory.go [hunk 3]
```diff
@@ -56,12 +55,15 @@ func IngredientByNameAndVersion(language, name, version string) (*IngredientAndV
...
 		}
-		v := ingredient.Version.Version
-		if v != nil && *v == version {
-			return &IngredientAndVersion{
-				ingredient.V1IngredientAndVersion,
-				ingredient.Namespace,
-			}, nil
+
+		for _, ver := range ingredient.Versions {
+			v := ver.Version
```
Comment (MDrakos) [inline pkg/platform/model/inventory.go:60]: This assignment seems unnecessary

**Review by MDrakos — APPROVED** [review]
LGTM. Just a few nits

File: pkg/platform/model/inventory.go (hunk 1 above)
Comment (mdrohmann) [reply pkg/platform/model/inventory.go:63]: I will make this more obvious. :)

**Review by MDrakos — APPROVED** [review]

**Pull request merged**